libraries/botbuilder-dialogs/src/memory/componentMemoryScopes.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * @module botbuilder-dialogs
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { tests } from 'botbuilder-stdlib';
+import { MemoryScope } from './scopes';
+
+/**
+ * Interface for enumerating memory scopes.
+ */
+export interface ComponentMemoryScopes {
+    getMemoryScopes(): MemoryScope[];
+}
+
+/**
+ * Check if a [ComponentRegistration](xref:botbuilder-core.ComponentRegistration) is
+ * [ComponentMemoryScopes](xref:botbuilder-dialogs.ComponentMemoryScopes) or not.
+ *
+ * @param {ComponentRegistration} component The component registration.
+ * @returns {boolean} Type check result.
+ */
+export function isComponentMemoryScopes(component: any): component is ComponentMemoryScopes {
+    return tests.isObject(component) && tests.isFunc(component['getMemoryScopes']);
```

Comment:
If you want a fun(?) problem - try to write a `tests.unsafe.isObjectAs<T>(val: any): val is T` function that mirror the `assert.unsafe.castObjectAs<T>` function. It's a bit of a hack, but as long as it's combined with other tests/assertions I've found it to be helpful. Then you could write:

```typescript
export const isComponentMemoryScopes: Test<ComponentMemoryScopes> = (val) => {
  return tests.unsafe.isObjectAs<ComponentMemoryScopes>(component) && tests.isFunc(component.getMemoryScopes);
};
```

Not critical, just might be nice to have around.

Reply:
Thanks, I've tried the following implementation and it works, is it okay to include this in `botbuilder-stdlib`?
```
function isObjectAs<T>(val: unknown): val is T {
    if (isObject(val)) {
        castObjectAs<T>(val, []);
        return true;
    }
    return false;
}
```